common/src/api/internal/nexus.rs

```diff
@@ -110,20 +110,30 @@ pub struct UpdateArtifactId {
 //
 // 1. Add it here.
 //
-// 2. Add it by hand to <repo root>/openapi/{nexus-internal.json,sled_agent.json}
-//    (search for `"KnownArtifactKind"` with double-quotes).
+// 2. Add the new kind to <repo root>/{nexus-client,sled-agent-client}/lib.rs.
+//    The mapping from `UpdateArtifactKind::*` to `types::UpdateArtifactKind::*`
+//    must be left as a `todo!()` for now; `types::UpdateArtifactKind` will not
+//    be updated with the new variant until step 5 below.
 //
-//    TODO: is there a better way to do this? Rain couldn't figure out how to run
-//    the command to regenerate the API: the nexus build kept failing. There
-//    might be a circular dependency here, which is worrisome.
-//
-// 3. Add the new kind to <repo root>/{nexus-client,sled-agent-client}/lib.rs.
-//
-// 4. Add it to <repo root>/common/src/sql/dbinit.sql under (CREATE TYPE
+// 3. Add it to <repo root>/common/src/sql/dbinit.sql under (CREATE TYPE
 //    omicron.public.update_artifact_kind).
 //
 //    TODO: After omicron ships this would likely involve a DB migration.
 //
+// 4. Add the new kind and the mapping to its `update_artifact_kind` to
+//    <repo root>/nexus/db-model/src/update_artifact.rs
+//
+// 5. Regenerate the OpenAPI specs for nexus and sled-agent:
+//
+//    ```
+//    EXPECTORATE=overwrite cargo test test_nexus_openapi_internal
+//    EXPECTORATE=overwrite cargo test test_sled_agent_openapi_sled
```

Comment:
This can be a single command:

```
EXPECTORATE=overwrite cargo test -p omicron-nexus -p omicron-sled-agent test_nexus_openapi_internal test_sled_agent_openapi_sled
```

Or just:

```
EXPECTORATE=overwrite cargo test -p omicron-nexus -p omicron-sled-agent openapi
```